lib/otel/span-processor.js

```diff
@@ -97,44 +95,39 @@ module.exports = class NrSpanProcessor {
 
       if (key === ATTR_HTTP_REQUEST_METHOD || key === ATTR_HTTP_METHOD) {
         key = 'procedure'
+        segment.addAttribute(key, sanitized)
       } else if (key === ATTR_SERVER_ADDRESS || key === ATTR_NET_PEER_NAME) {
         key = 'hostname'
         sanitized = this.sanitizeHostname(sanitized)
         segment.addSpanAttribute(key, sanitized)
-      } else if (key === ATTR_HTTP_HOST) {
-        key = 'host'
-        sanitized = this.sanitizeHostname(sanitized)
       } else if (key === ATTR_SERVER_PORT || key === ATTR_NET_PEER_PORT) {
         key = 'port'
         segment.addSpanAttribute(key, sanitized)
-      } else if (key === ATTR_HTTP_RESP_STATUS_CODE || key === ATTR_HTTP_STATUS_CODE) {
+      } else if (key === ATTR_HTTP_RES_STATUS_CODE || key === ATTR_HTTP_STATUS_CODE) {
         key = 'http.statusCode'
         segment.addSpanAttribute(key, sanitized)
       } else if (key === ATTR_HTTP_STATUS_TEXT) {
         key = 'http.statusText'
         segment.addSpanAttribute(key, sanitized)
-      } else if (key === ATTR_URL_SCHEME || key === ATTR_HTTP_URL) {
-        sanitized = this.constructUrl(span)
+      } else if (key === ATTR_FULL_URL || key === ATTR_HTTP_URL) {
+        sanitized = urltils.scrubAndParseParameters(sanitized)
+        const path = urltils.obfuscatePath(this.agent.config, sanitized.path)
+        const host = this.sanitizeHostname(sanitized.host)
+        sanitized = `${sanitized.protocol}//${host}${path}`
         key = 'url'
-      } else if (key === ATTR_URL_QUERY || key === ATTR_HTTP_TARGET) {
+        segment.addAttribute(key, sanitized)
+      } else if (key === ATTR_URL_QUERY || key === ATTR_FULL_URL || key === ATTR_HTTP_URL) {
         this.addQueryParameters(segment, sanitized)
+      } else {
+        segment.addAttribute(key, sanitized)
       }
-      segment.addAttribute(key, sanitized)
     }
   }
 
   sanitizeHostname(hostname) {
     return urltils.isLocalhost(hostname) ? this.agent.config.getHostnameSafe(hostname) : hostname
   }
 
-  constructUrl(span) {
-    const scheme = span.attributes[ATTR_URL_SCHEME] || span.attributes[ATTR_HTTP_URL]?.split(':')[0]
-    const host = span.attributes[ATTR_SERVER_ADDRESS] || span.attributes[ATTR_NET_PEER_NAME]
-    let path = urltils.scrubAndParseParameters(span.attributes[ATTR_URL_PATH] || span.attributes[ATTR_HTTP_TARGET])
-    path = urltils.obfuscatePath(this.agent.config, path)
-    return `${scheme}://${host}${path.path}`
-  }
-
   addQueryParameters(segment, query) {
```

Comment:
looks like you lack until tests adding query params and asserting they are set on segment

Reply:
Added to test